### PR TITLE
Adding better types

### PR DESCRIPTION
**Pull request by mamazu**
Using phpactor to generate return types for phpactor code.

What is in here:
* Specifying types and reducing the amount of "fromUnknown" calls since now we know what types there are.
* Removing deprecated `isClass` and `isInterface` calls from the code and replaced them with instanceof calls.
* Moving one of the tests from array returns to Generator (and include the tests).

### CODE REVIEW

File: lib/Filesystem/Domain/MappedFilesystemRegistry.php [hunk 1]
```diff
@@ -35,7 +35,6 @@ public function has(string $name): bool
         return isset($this->filesystems[$name]);
     }
 
-    /** @return array<string> */
```
Comment (mamazu) [inline lib/Filesystem/Domain/MappedFilesystemRegistry.php:38]: Moved to the interface.

File: lib/Filesystem/Domain/FilePath.php [hunk 2]
```diff
@@ -54,19 +57,6 @@ public static function fromUnknown($path): FilePath
...
-        }
-
-        if ($path instanceof SplFileInfo) {
-            return self::fromSplFileInfo($path);
-        }
-
-        throw new RuntimeException(sprintf(
-            'Do not know how to create FilePath from "%s"',
-            is_object($path) ? get_class($path) : gettype($path)
-        ));
```
Comment (mamazu) [inline lib/Filesystem/Domain/FilePath.php:69]: Those were only used in tests. The rest of the code already uses the correct function.

Comment (dantleech) [reply lib/Filesystem/Domain/FilePath.php:69]: they were used by tests though, removing this changes the purpose of the function (`fromUnknown`)

Comment (dantleech) [reply lib/Filesystem/Domain/FilePath.php:69]: can rename to `fromFilePathOrString` for clarity

Comment (mamazu) [reply lib/Filesystem/Domain/FilePath.php:69]: If possible I would like to remove those kinds of methods entirely and reduce the amount of "unknown"s from phpactor. :D

**Comment by dantleech** [comment]
thanks

**Pull request merged**